### ISSUE TITLE
New feature: Add ability to clear notifications

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/community-app#4698]
Users should be able dismiss/clear notifications that they have read.

<img width="385" alt="Screenshot 2020-07-30 at 8 29 13 PM" src="https://user-images.githubusercontent.com/58783823/88938779-5ec9b100-d2a3-11ea-8283-37ff3ced50ad.png">

**Comment by Oanh-and-only-Oanh** [comment]
@Dara-K can you help add UI to allow user to clear their notifications?

**Comment by Dara-K** [comment]
Here is the Clear All Option: https://marvelapp.com/prototype/dcc2h8h/screen/71648003/handoff

**Comment by luizrrodrigues** [comment]
Hey @rootelement before we able to implement this in FE, we'll need add new feature to Notification API:

1. Endpoint to dismiss a notification or multiple notification items;
2. Able to filter by `dismiss=false`

**Comment by luizrrodrigues** [comment]
Hey @sachin-maheshwari before we able to implement this in FE, we'll need add new feature to Notification API:

1. Endpoint to dismiss a notification or multiple notification items;
2. Able to filter by `dismiss=false`

**Comment by sachin-maheshwari** [comment]
@luizrrodrigues we have endpoint PUT https://api.topcoder-dev.com/v5/notifications/<id>/read , which can clear notification logically (by changing "read" flag to 1 in the backend and at time of fectching we need to give like https://api.topcoder-dev.com/v5/notifications/?platform=community&**read=false**&limit=20)
Hopefully it will help.

**Comment by SathyaJayabal** [comment]
@sachin-maheshwari @luizrrodrigues , we have two separate  things to track from the FE, read and dismiss. Read will remove the dot next to the notification and dismiss will remove the notification from the popup.

cc @Oanh-and-only-Oanh

**Comment by sachin-maheshwari** [comment]
@SathyaJayabal I think at Connect, it is managing by "seen" and "read" flag. Please check one time at Connect.

**Comment by SathyaJayabal** [comment]
@sachin-maheshwari , I checked in connect, in connect when u mark as read, then the notification disappears, but in  community when  u mark as read only the dot next to it disappears. We had a cross icon next to the notification to dismiss the notification.

@Oanh-and-only-Oanh , we need to decide on the UX before we go forward.

**Comment by luizrrodrigues** [comment]
@SathyaJayabal Thanks for additional information.

@sachin-maheshwari I was thinking about using the `seen` or `read` to control this in the popup, but if we look at the design, this is more like an additional feature.

So we need to follow this flow:
- When opening the notification popup, an item with `seen=false` will show a red mark. FE will call `/seen` to mark visible notifications as read (so next time open popup red mark will be removed)
- When a notification item already is seen before, but not read yet, will show a grey mark
- If click on the notification item, FE will call `/read` and remove the grey mark, but the notification will keep in the popup
- (new feature) If already read before, instead of a grey mark we'll show a dismiss icon 'X', if click in dismiss icon will remove the notification item from the popup
- (new feature) Also we able to click in 'Clear All' on top of the popup to remove all notifications

**Comment by luizrrodrigues** [comment]
@sachin-maheshwari Any update in this one?

**Comment by sachin-maheshwari** [comment]
@SathyaJayabal, why there are two different UX. Still unable to understand, either we should fix the UX in Connect too ?

**Comment by sachin-maheshwari** [comment]
Look at this for other organization. Initial blue color suggesting new notifications
<img width="1392" alt="Screenshot 2020-09-15 at 12 12 03 AM" src="https://user-images.githubusercontent.com/6467888/93125449-c79ca980-f6e8-11ea-9af8-d753d27f289c.png">
 are available.

**Comment by sachin-maheshwari** [comment]
Once i clicked, its blue color disappear ("seen"). 
<img width="1488" alt="Screenshot 2020-09-15 at 12 12 18 AM" src="https://user-images.githubusercontent.com/6467888/93125585-ff0b5600-f6e8-11ea-9e1a-bbc2c98fcff8.png">

**Comment by sachin-maheshwari** [comment]
There is cross icon to dismiss the notification ("read"). 
<img width="1077" alt="Screenshot 2020-09-15 at 12 12 52 AM" src="https://user-images.githubusercontent.com/6467888/93125765-3974f300-f6e9-11ea-836e-d0f59cc93e59.png">

**Comment by sachin-maheshwari** [comment]
So only two cases are possible - seen and read (dismiss).

**Comment by Oanh-and-only-Oanh** [comment]
@sachin-maheshwari, without the ability to remove or clear all the notifications from the popup, it starts to become very cluttered.

**Comment by sachin-maheshwari** [comment]
@Oanh-and-only-Oanh We can user this endpoint PUT https://api.topcoder-dev.com/v5/notifications/{id}/read  to remove the notifications

**Comment by Oanh-and-only-Oanh** [comment]
we want the user to clear/remove all notifications in the drop down regardless if it's is "seen" or "read".

**Comment by sachin-maheshwari** [comment]
PUT https://api.topcoder-dev.com/v5/notifications/read will remove all notifications.

**Comment by sachin-maheshwari** [comment]
only we need to take care at time of fetching (or making notification list) , we should use this call - GET https://api.topcoder-dev.com/v5/notifications/?platform=community&read=false&limit=20)

**Comment by sachin-maheshwari** [comment]
Here is only confusion with name of the endpoint i.e. "read", would be more appropriate if we use "dismiss". Otherwise functionality wise its same.

**Comment by luizrrodrigues** [comment]
@sachin-maheshwari yeah, I see your point here, but if we use 'read' to dismiss, we'll lose ability to read a notification item and keep this in popup.

Example:
- I see notification about a event;
- Click to see details
- Is a big text, so want read this late
- when back to read this, I go to popup, but notification event is gone, so need click in "View All Notifications"


@Oanh-and-only-Oanh if scenario above isn't a problem, we can go forward and use 'read'

**Comment by Oanh-and-only-Oanh** [comment]
@luizrrodrigues, you're correct that would create a problem we don't want.
